nexus/types/src/deployment.rs

```diff
@@ -828,3 +794,484 @@ pub struct UnstableReconfiguratorState {
+            let blueprint_zones = &self.blueprint.blueprint_zones;
+            let mut builder = ZoneTableBuilder::new(ZONE_INDENT.to_string());
+
+            for (sled_id, sled_zones) in blueprint_zones {
+                // Add a heading row.
+                builder.push_heading(format!(
+                    "{SLED_INDENT}sled {sled_id}: zones at generation {}",
+                    sled_zones.generation
+                ));
+
+                // Add a row for each zone.
+                for zone in &sled_zones.zones {
+                    builder.push_zone_record(ZONE_INDENT.to_string(), zone);
+                }
+            }
+
+            builder.build()
+        }
+    }
+
+    impl<'diff, 'a> OmicronZonesDiffDisplay<'diff, 'a> {
+        pub(super) fn make_zones_diff_table(&self) -> Table {
+            let diff = self.diff;
+
+            // Add the unchanged prefix to the zone indent since the first
+            // column will be used as the prefix.
+            let mut builder = ZoneTableBuilder::new(format!(
+                "{UNCHANGED_PREFIX}{ZONE_INDENT}"
+            ));
+
+            // Print out the whole sled for removed sleds.
+            for (sled_id, sled_zones) in diff.sleds_removed() {
+                self.add_whole_sled_records(
+                    sled_id,
+                    sled_zones,
+                    WholeSledKind::Removed,
+                    &mut builder,
+                );
+            }
+
+            // For sleds that are in common:
+            for (sled_id, sled_changes) in diff.sleds_in_common() {
+                // Print a line about the sled itself and zone config
+                // generation, regardless of whether anything has changed.
+                if sled_changes.generation_before
+                    != sled_changes.generation_after
+                {
+                    builder.push_heading(format!(
+                        "{CHANGED_PREFIX}{SLED_INDENT}sled {sled_id}: \
+                         zones at generation {} -> {}",
+                        sled_changes.generation_before,
+                        sled_changes.generation_after,
+                    ));
+                } else {
+                    builder.push_heading(format!(
+                        "{UNCHANGED_PREFIX}{SLED_INDENT}sled {sled_id}: zones at generation {}",
+                        sled_changes.generation_before,
+                    ));
+                }
+
+                for (i, zone) in sled_changes.zones_removed().enumerate() {
+                    if i == 0 {
+                        builder.push_subheading(format!(
+                            "{REMOVED_PREFIX}{SLED_SUBHEAD_INDENT}removed zones:"
+                        ));
+                    }
+                    builder.push_zone_record(
+                        format!("{REMOVED_PREFIX}{ZONE_INDENT}"),
+                        zone,
+                    );
+                }
+
+                // First print out all changed zones.
```

Comment:
Tiny nit - this is second after printing all removed zones

Reply:
Fixed, thanks.